### PR TITLE
Plant analyzer can scan the health of podmen.

### PR DESCRIPTION
**Pull request by KathyRyals**
## About The Pull Request

Makes it so you can perform a health scan / chemical scan on a podperson with a health analyzer.

## Why It's Good For The Game

It's a small flavour change, and I think it'll make for a cool addition.

## Changelog
:cl: Kathy Ryals
add: The plant analyzer has been updated to be able to analyze a Podman health or chemical contents.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/hydroitemdefines.dm [hunk 1]
```diff
@@ -18,6 +18,17 @@
...
+	//Checks if target is a podman
+	if(ispodperson(M))
+		user.visible_message("<span class='notice'>[user] analyzes [M]'s vitals.</span>", \
+							"<span class='notice'>You analyze [M]'s vitals.</span>")
+		if(scan_mode== PLANT_SCANMODE_STATS)
+			healthscan(user, M, advanced = TRUE)
+		else
+			chemscan(user, M)
+	add_fingerprint(user)
+
```
Comment (Rohesie) [inline code/modules/hydroponics/hydroitemdefines.dm:31]:
```suggestion
	if(ispodperson(M))
		user.visible_message("<span class='notice'>[user] analyzes [M]'s vitals.</span>", \
							"<span class='notice'>You analyze [M]'s vitals.</span>")
		if(scan_mode== PLANT_SCANMODE_STATS)
			healthscan(user, M, advanced = TRUE)
		else
			chemscan(user, M)
		add_fingerprint(user)
		return
	return ..()

```

**Comment by Rohesie** [comment]
Make it so non-advanced health scanners can't scan podpeople if they can be scanned with advanced results by the plant analyzers.

Trade-offs.

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -127,10 +127,14 @@ GENE SCANNER
...
 
+	if(ispodperson(M)&& !advanced)
+		to_chat(user, "<span class='info'>[M]'s biologal structure is too complicated for the health analyzer.")
+		return
+
 	user.visible_message("<span class='notice'>[user] analyzes [M]'s vitals.</span>", \
 						"<span class='notice'>You analyze [M]'s vitals.</span>")
 
-	if(scanmode == SCANMODE_HEALTH)
+	if((scanmode == SCANMODE_HEALTH))
```
Comment (Rohesie) [inline code/game/objects/items/devices/scanners.dm:137]:
```suggestion
	if(scanmode == SCANMODE_HEALTH)
```
Why the double parentheses?

Comment (KathyRyals) [reply code/game/objects/items/devices/scanners.dm:137]: Remains from an earlier try !

Comment (KathyRyals) [reply code/game/objects/items/devices/scanners.dm:137]: Will merge asap

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -127,6 +127,10 @@ GENE SCANNER
 					 \n<span class='info'>Body temperature: ???</span>")
 		return
 
+	if(ispodperson(M)&& !advanced)
```
Comment (ATH1909) [inline code/game/objects/items/devices/scanners.dm:130]: does this affect those pay-per-usse health scanners that no one ever uses?

Comment (KathyRyals) [reply code/game/objects/items/devices/scanners.dm:130]: I'll check ! I don't this particulat method is used outside of health analyzers and PDA though.

Comment (KathyRyals) [reply code/game/objects/items/devices/scanners.dm:130]: It works as intended.

**Review by Rohesie — APPROVED** [review]

**Pull request merged**